docs/v1/api.md

```diff
@@ -0,0 +1,335 @@
+- Ensure a smooth, consistent UX for users of the API and implementors of the
+  controllers
+
+# API Precedents in Kubernetes
+
+Broadly speaking, the system detailed herein can list services and allow
+consumers to provision, bind, unbind and deprovision them.
+
+We've drawn inspiration from the
+[Persistent Volumes](http://kubernetes.io/docs/user-guide/persistent-volumes/)
+API and the [Ingress Controllers](http://kubernetes.io/docs/user-guide/ingress/)
+system. The remainder of this section explains these inspirations in further
+detail.
+
+Additionally, we've adhered, wherever possible, to Kubernetes's declarative
+API. That is, the resources we've defined herein describe the _desired end
+state_ of the applicable systems. Kubernetes and the service catalog controller
+implementations (see below) are responsible for fulfilling that end state.
+
+## Persistent Volumes
+
+First, this problem space is similar to that of persistent volumes. We model
+the data structures herein after those in persistent volumes primarily because
+we believe that consistency is a virtue. More importantly, the community
+benefits when similar problems have similar solutions.
+
+Specifically, we believe that the following existing types provide a good model
+for the mechanics of how the services offered by a broker can be used in
+Kubernetes:
+
+- `StorageClass`
+- `PersistentVolume`
+- `PersistentVolumeClaim`
+
+## Ingress Controllers
+
+Today, operators choose their own implementation of an ingress controller.
+Kubernetes gives the operator the ability to specify any ingress controller,
+and the complete freedom to implement it themselves.
+
+Similarly, permitting operators to implement or select their own service catalog
+controller would encourage freedom of choice and diversity. The analog to an
+ingress controller for the service catalog system is called a _service catalog
+controller_. Like ingress, these controllers run inside a standard pod. Anyone
+can write and deploy a service catalog controller, with no changes necessary to
+the Kubernetes install.
+
+Also as in the case of ingress controllers, a cluster may contain many
+different service catalog controllers, each of which handles operations for a
+designated subset of the applicable resource type. For example, a cluster may
+contain a service catalog controller to handle AWS RDS databases, and a
+completely separate service catalog controller to handle on-cluster Redis
+instances.
+
+As we'll detail below, each service catalog controller implementation must
+conform to an implementation spec, but that spec allows for a heterogenous set
+of controllers in a cluster. We believe this allowance encourages flexibility
+in a cluster's service catalog.
+
+# Kubernetes Resources
+
+Given these inspirations, the following resources will be added to the
+Kubernetes API:
+
+- `ServiceClass`
+  - Describes a _kind_ of service. This resource is semantically similar to how
```

Comment:
I'd suggest we remove all of this text about volumes - it just confuses things - especially when we end up having more text comparing ServiceClasses and Volumes rather than just saying what ServiceClasses are.
Proposal:
Describes a _kind_ of service that is available from the Catalog. Each type of service, irrespective of the exact features it provides, will have a corresponding ServiceClass.  For example, if the Catalog offers a MySQL and a MongoDB service, then there will be two ServiceClasses even though they are both database types of services.